controller/engine_controller.go

```diff
@@ -1120,6 +1122,26 @@ func (ec *EngineController) syncSnapshotCRs(engine *longhorn.Engine) error {
 	return nil
 }
 
+func IsValidForExpansion(engine *longhorn.Engine, cliAPIVersion int) (bool, error) {
+	if engine.Status.IsExpanding {
+		return false, nil
+	}
+	if engine.Spec.VolumeSize == engine.Status.CurrentSize {
+		return false, nil
+	}
+	if engine.Spec.VolumeSize < engine.Status.CurrentSize {
+		return false, fmt.Errorf("BUG: The expected size %v of engine %v should not be smaller than the current size %v", engine.Spec.VolumeSize, engine.Name, engine.Status.CurrentSize)
+	}
+	if cliAPIVersion < engineapi.MinCLIVersion {
+		return false, nil
+	}
+	if cliAPIVersion < 7 && engineapi.IsEndpointTGTBlockDev(engine.Status.Endpoint) {
+		return false, fmt.Errorf("cannot do online expansion for the old engine")
```

Comment:
Suggest seperate these conditions, because we will only support engine running with block device.

```go

if !engineapi.IsEndpointTGTBlockDev(engine.Status.Endpoint) {
...
}

if cliAPIVersion < 7 {
}
```

Reply:
> we will only support engine running with block device.
We will support expanding block device when the cli API version >= 7